transform-plugins/src/main/java/io/cdap/plugin/NullFieldSplitter.java

```diff
@@ -139,10 +145,43 @@ static Schema getNonNullSchema(Schema nullableSchema, String fieldName) {
     return Schema.recordOf(nullableSchema.getRecordName() + ".nonnull", fields);
   }
 
+  @VisibleForTesting
+  static Schema getNonNullSchema(Schema nullableSchema, String fieldName, FailureCollector collector) {
```

Comment:
shouldn't this replace the other method?

Reply:
The reason I made an overload is that the SplitterTransform does not contain a getContext() method necessary to get a failure collector, and getNonNullSchema is also used in the transform method, so there's no easy way to pass a failure collector in that context.

Reply:
I see. Since these methods are basically the same, let's make the FailureCollector nullable and pass in a null instead of calling the other one.

Reply:
Sounds good, fixed